package/MDAnalysis/lib/pkdtree.py

```diff
@@ -145,75 +146,87 @@ def set_coords(self, coords):
+    def find_images(self, center_points, radius):
         """
-        Find relevant images of a center point, inspired by
+        Find relevant images of a a set of center points, inspired by
         https://github.com/patvarilly/periodic_kdtree
 
         Parameters
         ----------
-        center_point: NumPy.array
-          Coordinates of the query center point
+        center_points: NumPy.array shape=(N,3)
+          Coordinates of the query center points. Must be in centrall cell.
         radius: float
           Maximum distance from center in search for neighbors
 
         Returns
         ------
         :class:`List`
-          wrapped center point and its relevant images
+          images of the center points close to any cell boundary plane
         """
-        wrapped_c = apply_PBC(center_point.reshape(1, 3), self.box)[0]
-        # extents marks the max distance to plane just to consider an image
-        extents = np.array([norm(self._dm[i])/2.0 for i in range(self.dim)])
-        extents = np.where(extents < radius, extents, radius)
-        # displacements are vectors that we add to wrapped_c to
-        # generate images "up" or "down" the central cell along
-        # the axis that we happen to be looking.
-        displacements = list()
-        for i in range(self.dim):
-            # distance to the plane containing the origin
-            if np.dot(wrapped_c, self._rm[i]) < extents[i]:
-                displacements.append(self._dm[i])  # "upper" image
-            # distance to the plane containing point self._dm[i]
-            elif np.dot(self._dm[i] - wrapped_c, self._rm[i]) < extents[i]:
-                displacements.append(-self._dm[i])  # "lower" image
-        # If we have displacements along more than one axis, we have
-        # to combine them. This happens when wrapped_center is close
-        # to any edge or vertex of the central cell.
-        # face, n_displacements==1; no combination
-        # edge, n_displacements==2; combinations produce one extra displacement
-        # vertex, n_displacements==3; five extra displacements
-        n_displacements = len(displacements)
-        if n_displacements > 1:
-            for start in range(n_displacements - 1, -1, -1):
-                for i in range(start+1, len(displacements)):
-                    displacements.append(displacements[start]+displacements[i])
-        return [wrapped_c, ] + [wrapped_c + d for d in displacements]
-
-    def search(self, center, radius):
-        """Search all points within radius of center and its periodic images.
+        images = list()
+
+        if center_points.shape == (3,):
+            center_points = center_points.reshape((1,3))
+
+        # To-Do: parallel
+        for center_point in center_points:
+            # displacements are vectors we add to center_point to generate
+            # images "up" or "down" the central cell along the axis that we
+            # happen to be looking.
```

Comment:
Calculate displacement vectors for images of `center_point` in neighboring boxes.